src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -2317,6 +2317,9 @@ private void PopulateFieldRvaTableRows(BlobBuilder mappedFieldDataWriter)
                 }
 
                 int offset = mappedFieldDataWriter.Count;
+                Debug.Assert(offset % ManagedPEBuilder.MappedFieldDataAlignment == 0, "Expected last write to end at alignment boundary");
+                Debug.Assert(ManagedPEBuilder.MappedFieldDataAlignment == 8, "Expected alignment to be 8");
```

Comment:
nit: this may be a good location to document the fact that we emit RVAs as multiples of 8, but packingSize/.pack/Alignment emitted by `PopulateClassLayoutTableRows` is finer-grained. Alternatively, in `PopulateClassLayoutTableRows` could be good too.